archetypes/embedded/src/main/resources/archetype-resources/src/main/java/Application.java

```diff
@@ -0,0 +1,122 @@
+
+      assert oldValue == null;
+      assert worked == true;
+   }
+
+   public void lifespans() throws InterruptedException {
+      System.out.println("\n\n2.  Demonstrating usage of Infinispan with expirable entries.");
+      Cache<String, Float> stocksCache = SampleCacheContainer.getCache("stockTickers");
+      System.out.println("  Storing key 'RHT' for 10 seconds.");
+      stocksCache.put("RHT", 45.0f, 10, TimeUnit.SECONDS);
+      System.out.printf("  Checking for existence of key.  Is it there? %s\n", stocksCache.containsKey("RHT"));
+      System.out.println("  Sleeping for 10 seconds...");
+      Thread.sleep(10000);
+      System.out.printf("  Checking for existence of key.  Is it there? %s\n", stocksCache.containsKey("RHT"));
+      assert stocksCache.get("RHT") == null;
+   }
+
+   public void asyncOperations() {
+      System.out.println("\n\n3.  Demonstrating asynchronous operations - where writes can be done in a non-blocking fashion.");
+      Cache<String, Integer> wineCache = SampleCacheContainer.getCache("wineCache");
+
+      System.out.println("  Put #1");
+      CompletableFuture<Integer> f1 = wineCache.putAsync("Pinot Noir", 300);
+      System.out.println("  Put #1");
+      CompletableFuture<Integer> f2 = wineCache.putAsync("Merlot", 120);
+      System.out.println("  Put #1");
+      CompletableFuture<Integer> f3 = wineCache.putAsync("Chardonnay", 180);
+
+      // now poll the futures to make sure any remote calls have completed!
+      for (CompletableFuture<Integer> f : Arrays.asList(f1, f2, f3)) {
+         try {
+            System.out.println("  Checking future... ");
+            f.get();
+         } catch (Exception e) {
+            throw new RuntimeException("Operation failed!", e);
+         }
+      }
+      System.out.println("  Everything stored!");
+
+      // TIP: For more examples on using the asynchronous API, visit https://infinispan.org/docs/stable/titles/developing/developing.html#cache_asynchronous_api
+   }
+
+   public void registeringListeners() {
+      System.out.println("\n\n4.  Demonstrating use of listeners.");
+      Cache<Integer, String> anotherCache = SampleCacheContainer.getCache("another");
+      System.out.println("  Attaching listener");
+      MyListener l = new MyListener();
+      anotherCache.addListener(l);
+
+      System.out.println("  Put #1");
+      anotherCache.put(1, "One");
+      System.out.println("  Put #2");
+      anotherCache.put(2, "Two");
+      System.out.println("  Put #3");
+      anotherCache.put(3, "Three");
+
+      // TIP: For more examples on using listeners visit https://infinispan.org/docs/stable/titles/developing/developing.html#listeners_and_notifications
+   }
+
+   public static void main(String[] args) throws Exception {
+      System.out.println("\n\n\n   ********************************  \n\n\n");
+      System.out.println("Hello.  This is a sample application making use of Infinispan.");
+      Application a = new Application();
```

Comment:
I would move the instantiation before the `println` and create/invoke a `star()` method. Just to remove the log4j entries out of the way
```
1.  Demonstrating basic usage of Infinispan.  This cache stores arbitrary Strings.
Jun 17, 2020 6:19:48 PM org.infinispan.configuration.parsing.Parser parseStoreAsBinary
WARN: ISPN000428: The 'store-as-binary' element has been deprecated. Please use the 'memory' element instead
Jun 17, 2020 6:19:48 PM org.infinispan.configuration.parsing.Parser parseMemory
WARN: ISPN000586: The memory configuration element 'object' has been deprecated. Please update your configuration
Jun 17, 2020 6:19:49 PM org.infinispan.factories.GlobalComponentRegistry preStart
INFO: ISPN000128: Infinispan version: Infinispan 'Corona Extra' 11.0.1-SNAPSHOT
Jun 17, 2020 6:19:49 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport start
INFO: ISPN000078: Starting JGroups channel mycluster with stack tcp
Jun 17, 2020 6:19:51 PM org.jgroups.protocols.pbcast.ClientGmsImpl joinInternal
INFO: pedro-laptop-3-14776: no members discovered after 2002 ms: creating cluster as coordinator
Jun 17, 2020 6:19:51 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport receiveClusterView
INFO: ISPN000094: Received new cluster view for channel mycluster: [pedro-laptop-3-14776|0] (1) [pedro-laptop-3-14776]
Jun 17, 2020 6:19:51 PM org.infinispan.remoting.transport.jgroups.JGroupsTransport startJGroupsChannelIfNeeded
INFO: ISPN000079: Channel mycluster local address is pedro-laptop-3-14776, physical addresses are [172.17.0.1:7800]
  Storing value 'World' under key 'Hello'
  Done.  Saw old value as 'null'
  Replacing 'World' with 'Mars'.
  Successful? true
```

Reply:
I have removed the `SampleCacheContainer` in favour of keeping everything within `Application.java`. The `Application` class is now instantiated and ran inside of a `try(EmbeddedCacheManager){}` block